TLS test failures

mrocklin opened the pull request:
Do not merge

mrocklin commented:
@jcrist @mariusvniekerk do either of you have time to check out these test failures?  They're breaking tests currently.

mrocklin commented:
Also cc @pitrou

pitrou commented:
You should be able to pinpoint when the failures started to occur? Is it on a specific changeset?

mrocklin commented:
I haven't tried doing this yet.  They didn't show up during any PR and now they fail consistently, so I don't think that they snuck in as some intermittent failures do. 

My guess is that some dependency changed upstream.

pitrou commented:
Do you `conda list` as part of CI jobs? That's very useful for cases like this, so that you can compare versions.

pitrou commented:
By the way, I don't think those failures are intermittent. They should even be reproducible locally, with the right software setup.

mariusvniekerk commented:
@pitrou Pretty certain these are related to openssl 1.1 vs 1.0

pitrou commented:
@mariusvniekerk do you have a proof of that? e.g. `conda list` output?

mariusvniekerk commented:
it does "warn" about it failing, so it is aware of that but doesn't raise?

pitrou commented:
Right, but it should raise in that case. The warning is in the case the handshake error is because of transient network issues. But if the certificate is reject, that's a permanent error and it should raise an exception. So you must inspect the exception to decide what to do with it.

mariusvniekerk commented:
We should change the test to catch and reraise the exception if present to see what it is,  we could still be raising **something** but pytest just says it's not `EnvironmentError`.  Want to just rule that one out first.
I'm not sure if i have push rights to this pr

mrocklin commented:
Marius, you now have push rights to this PR

On Thu, Nov 8, 2018 at 8:48 AM Marius van Niekerk <notifications@github.com>
wrote:

> We should change the test to catch and reraise the exception if present to
> see what it is, we could still be raising *something* but pytest just
> says it's not EnvironmentError. Want to just rule that one out first.
> I'm not sure if i have push rights to this pr
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dask/distributed/pull/2331#issuecomment-436999432>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AASszFWlJhYF3riDybMex57SM2z5z1QSks5utDY4gaJpZM4YGD7B>
> .
>

mrocklin commented:
Tests seem to pass.  The Python 2 failure was due to the `print(..., flush=True)` syntax in debug code.

mrocklin commented:
To the extent that I'm able to judge this seems fine to me.  There were a couple of test failures but I think that they are intermittent and unrelated.

@mariusvniekerk is there any additional work that you think should be done here?

mariusvniekerk commented on distributed/comm/tcp.py:
```diff
@@ -329,6 +333,13 @@ def connect(self, address, deserialize=True, **connection_args):
             stream = yield client.connect(ip, port,
                                           max_buffer_size=MAX_BUFFER_SIZE,
                                           **kwargs)
+            # Under certain circumstances tornado will have a closed connnection with an error and not raise
```
@mrocklin I think this gets us working for now.
@pitrou I don't think tornado is tested a python built against openssl 1.1+ atm?

pitrou replied:
I don't know, but it's still worth reporting. I generally find it annoying when people don't report issues upstream.

mariusvniekerk replied:
Done

mrocklin commented:
I removed my initial dummy commit and added a couple small style changes.  I hope that now that the first commit belongs to @mariusvniekerk that the "Squash and merge" feature will appropriately assign ownership to him

mrocklin commented:
This does mean a force push though, so you may want to `git reset --hard` before pulling/pushing if there is any more to do
